### PR TITLE
Orleans facet property injection

### PR DESCRIPTION
**Pull request by jason-bragg**
Initial implementation of Orleans Facet System #3223  using property injection.

### CODE REVIEW

File: src/OrleansRuntime/Facet/GrainFacetPropertyInjector.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            private readonly List<Action<IGrainActivationContext>> propertyInjectors;
+
+            public Injector(Type type)
+            {
+                this.propertyInjectors = CreatePropertyInjectors(type).ToList();
+            }
+
+            public void InjectProperties(IGrainActivationContext grainContext)
+            {
+                propertyInjectors.ForEach(injector => injector.Invoke(grainContext));
```
Comment (ReubenBond) [inline src/OrleansRuntime/Facet/GrainFacetPropertyInjector.cs]: Using `foreach` here would avoid these two per-activation allocations: one for the closure to capture `grainContext` and one for the conversion from method to `Func<Action<IGrainActivationContext>>`.

File: test/Tester/Facet/StorageFeatureAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Orleans;
+using Orleans.Runtime;
+
+namespace Tester
+{
+    [AttributeUsage(AttributeTargets.Property)]
```
Comment (ReubenBond) [inline test/Tester/Facet/StorageFeatureAttribute.cs]: If these attributes weren't restricted to properties, then they could one day be used on ctor params,

Comment (jason-bragg) [inline test/Tester/Facet/StorageFeatureAttribute.cs]: Agreed, but we'd need some additional wiring.  Best to keep them constrained to the only supported pattern for now.  Do you disagree?

**Comment by jason-bragg** [comment]
Returning to constructor injection.  Closing.

**Pull request closed without merging**